test/blit_test.py

```diff
@@ -130,6 +130,87 @@ def blits(blit_list):
         if PRINT_TIMING:
             print(f"Surface.blits generator: {t1 - t0}")
 
+    def test_ublits(self):
+
+        NUM_SURFS = 255
+        PRINT_TIMING = 0
+        dst = pygame.Surface((NUM_SURFS * 10, 10), SRCALPHA, 32)
+        dst.fill((230, 230, 230))
+        blit_list = self.make_blit_list(NUM_SURFS)
+
+        def blits(blit_list):
+            for surface, dest in blit_list:
+                dst.blit(surface, dest)
+
+        from time import time
+
+        # tests for number of parameters
+        self.assertRaises(ValueError, dst.ublits)  # no params
+        self.assertRaises(ValueError, dst.ublits, blit_list)  # 1 param
+        self.assertRaises(ValueError, dst.ublits, blit_list, 0)  # 2 params
+
+        # tests for the blit_sequence parameter
+        self.assertRaises(ValueError, dst.ublits, [-1])
+        self.assertRaises(ValueError, dst.ublits, (-1,))
+        self.assertRaises(ValueError, dst.ublits, "str")
+        self.assertRaises(ValueError, dst.ublits, 1)
+
+        # tests for the special_flags parameter
+        self.assertRaises(
+            TypeError, dst.ublits, blit_list, -999, True
+        )  # not impl. flag
+        self.assertRaises(ValueError, dst.ublits, blit_list, 12.332, True)  # float flag
+        self.assertRaises(ValueError, dst.ublits, blit_list, None, True)
+        self.assertRaises(ValueError, dst.ublits, blit_list, [], True)
+        self.assertRaises(ValueError, dst.ublits, blit_list, (), True)
+        self.assertRaises(ValueError, dst.ublits, blit_list, "str", True)
+        self.assertRaises(ValueError, dst.ublits, blit_list, "str", True)
+
+        # tests for the doreturn parameter
+        self.assertRaises(ValueError, dst.ublits, blit_list, 0, "str")
+        self.assertRaises(ValueError, dst.ublits, blit_list, 0, [])
+        self.assertRaises(ValueError, dst.ublits, blit_list, 0, [1])
+        self.assertRaises(ValueError, dst.ublits, blit_list, 0, ())
+        self.assertRaises(ValueError, dst.ublits, blit_list, 0, (1,))
+
+        # tests for return value
+        self.assertEqual(dst.ublits(blit_list, 0, False), None)
```

Comment:
This test fails with a `TypeError: Source objects must be a Surface`. Problem is tricky because it really starts at line 159. there's some kind of undefined behaviour that transforms the `blit_list` into a infinite recursive list that's obviously not compatible with ublits. Can't really figure it out.

Reply:
Ok i found uou that something corrupts only the first surface of the first tuple transforming the sequence to:
`[(<NULL>, (80.36447577626312, 241.40381563470999)), ...]` (i put the dots but everything past this first element is ok)
by doing
```python
try:
    win.ublits(ublits_sequence, -999, 1)
except TypeError:
    print(ublits_sequence)
```
and if i do
```python
try:
    win.ublits(ublits_sequence, 12.3324, 1)
except ValueError:
    print(ublits_sequence)
```
the sequence is ok just like in the tests.

Reply:
I think i fixed it with the latest commit!